components/motor/gpio/motor_encoder_test.go

```diff
@@ -599,17 +595,17 @@ func TestMotorEncoderIncremental(t *testing.T) {
 		realMotor := th.RealMotor
 		motor := th.Motor
 
-		err := motor.goForInternal(context.Background(), 100, 1)
+		err := motor.goForInternal(context.Background(), 100, 1, 1, 1)
 		test.That(t, err, test.ShouldBeNil)
 		test.That(t, realMotor.Direction(), test.ShouldEqual, 1)
 
-		err = motor.goForInternal(context.Background(), -100, -1)
+		err = motor.goForInternal(context.Background(), -100, -1, 1, 1)
 		test.That(t, err, test.ShouldBeNil)
 		test.That(t, realMotor.Direction(), test.ShouldEqual, 1)
 
 		testutils.WaitForAssertion(t, func(tb testing.TB) {
 			tb.Helper()
-			test.That(tb, realMotor.PowerPct(), test.ShouldEqual, 1.0)
+			test.That(tb, realMotor.PowerPct(), test.ShouldBeGreaterThan, 0.0)
```

Comment:
this test was failing and I couldn't see a reason why greater than 0 isn't sufficient... thoughts?

Reply:
Was it using the old ramp rate? That might  be why.

I agree with your assessment here though.

Reply:
@JohnN193 as for what we talked about with this test, I think it's fine because a negative rpm and a positive goal position doesn't necessarily indicate the direction you're going I guess. I think in this example it would be like if with GoFor you had a negative rpm and a negative revolutions which gets you to a goalPos of positive 1, and the direction is also 1. I forgot that goForInternal doesn't know the revolutions

Reply:
so looking at `goForInternal` again, it seems like direction only gets used to specify the [initial spin](https://github.com/viamrobotics/rdk/pull/3671/files#diff-c995a39b5b86039be0b4c03ff28ef502f44068848c99b80628eeaa9c9cef734dR448) of the motor, which in turn sets the direction on the motors state(by having a power set)

so even though the math in rpm monitor would result in a power going in the opposite direction, because we set an initial power to the motors there will always be a positive power. Then we have the [logic check](https://github.com/viamrobotics/rdk/pull/3671/files#diff-c995a39b5b86039be0b4c03ff28ef502f44068848c99b80628eeaa9c9cef734dR265-R266) in makeAdjustments to basically always have the power match the direction